### PR TITLE
fix: three-valued logic for NullExpr, ExistsExpr and logical operators

### PR DESCRIPTION
**Pull request by sunby**
Implement proper three-valued logic (TRUE/FALSE/NULL) support:
issue: https://github.com/milvus-io/milvus/issues/46820

1. NullExpr/ExistsExpr: valid bits always true (results are deterministic)
2. LogicalUnaryExpr (NOT): NULL input produces NULL output
3. ConjunctExpr (AND/OR): F AND NULL=F, T OR NULL=T (short-circuit)

Add unit tests for NOT (IS NOT NULL), NOT (IS NULL), and combined expressions.

### CODE REVIEW

File: internal/core/src/common/ValueOp.h [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        // left_valid = left_valid & (right_valid | ~left_data)
+        left_valid.inplace_and(tmp, size);
+
+        // tmp = right_valid & ~right_data (reuse tmp)
+        tmp.set();
+        tmp.inplace_xor(right_data, size);   // tmp = ~right_data
+        tmp.inplace_and(right_valid, size);  // tmp = right_valid & ~right_data
+
+        // left_valid |= (right_valid & ~right_data)
+        left_valid.inplace_or(tmp, size);
```
[the remark this answers is not in the document]
Comment (sunby) [reply internal/core/src/common/ValueOp.h:76]: no function called inplace_and_not

**Review by zhengbuqian — APPROVED** [review]

File: internal/core/src/exec/expression/NullExpr.cpp [hunk 2]
```diff
@@ -119,8 +119,8 @@ PhyNullExpr::ExecVisitorImpl(OffsetVector* input) {
     if (expr_->op_ == proto::plan::NullExpr_NullOp_IsNull) {
         res.flip();
     }
-    auto res_vec =
-        std::make_shared<ColumnVector>(std::move(res), std::move(valid_res));
+    auto res_vec = std::make_shared<ColumnVector>(
+        std::move(res), TargetBitmap(valid_res.size(), true));
```
Comment (zhagnlu) [inline internal/core/src/exec/expression/NullExpr.cpp:123]: now valid all true? is that ok ?

Comment (sunby) [reply internal/core/src/exec/expression/NullExpr.cpp:123]: > now valid all true? is that ok ?

yes, For null expr, this bitmap means all result is valid.

File: internal/core/src/common/ValueOp.h [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        // left_data = left_data | right_data
+        left_data.inplace_or(right_data, size);
+    }
+
+    static size_t
+    TrueCount(ColumnVectorPtr res) {
+        TargetBitmapView data(res->GetRawData(), res->size());
+        TargetBitmapView valid_data(res->GetValidRawData(), res->size());
+        TargetBitmap tmp(data);
+        tmp.inplace_and(valid_data, res->size());
```
Comment (xiaofan-luan) [inline internal/core/src/common/ValueOp.h:137]: there seems to be a copy here? 

  static size_t TrueCount(ColumnVectorPtr res) {
      const uint64_t* data = reinterpret_cast<const uint64_t*>(res->GetRawData());
      const uint64_t* valid = reinterpret_cast<const uint64_t*>(res->GetValidRawData());

      size_t count = 0;
      size_t num_words = (res->size() + 63) / 64;

      for (size_t i = 0; i < num_words; i++) {
          count += __builtin_popcountll(data[i] & valid[i]); 
      }
      return count;
  }
should we simply use a simd instruction to do count here? make sure it's comptible with both arm and intel and FalseCount should do the same -> Zero copy + SIMD

File: internal/core/src/exec/expression/ConjunctExpr.cpp [hunk 4]
```diff
@@ -40,46 +41,32 @@ PhyConjunctFilterExpr::ResolveType(const std::vector<DataType>& inputs) {
...
-        ConjunctElementFunc<false> func;
-        return func(input_result, result);
+        return static_cast<int64_t>(result->size() - res_data.count());
     }
 }
 
 bool
 PhyConjunctFilterExpr::CanSkipFollowingExprs(ColumnVectorPtr& vec) {
-    if ((is_and_ && AllFalse(vec)) || (!is_and_ && AllTrue(vec))) {
+    if ((is_and_ && common::ThreeValuedLogicOp::TrueCount(vec) == 0) ||
```
Comment (xiaofan-luan) [inline internal/core/src/exec/expression/ConjunctExpr.cpp:67]: +    if (is_and_) {
  +        return common::ThreeValuedLogicOp::FalseCount(vec) == vec->size();
  +    } else {
  +        return common::ThreeValuedLogicOp::TrueCount(vec) == vec->size();
       } should it be something like this?
